### PR TITLE
Refactored, Improved and fixed typos for the entire UI category

### PR DESCRIPTION
**Pull request by W2Wizard**
Yes the entire category!

There are many changed files so I hope I did not miss something very obvious.
I tried my best to find as many grammar issues and typos while also trying to improve some of the content and updating it, so feel free to find any mistakes I may have made along the way.

For instance:

https://user-images.githubusercontent.com/63303990/105180560-0628f180-5b2b-11eb-8e5e-1d526f7e4257.mp4

I gave one tutorial some nice gifs, I made sure to reduce the file size of the gifs as good as I can, usually trying to keep it under 1MB. Will probably follow up with some more gifs if people like them.

In some cases code examples were not correct so I fixed them or optimized them a little bit, like here for example.

**From:**
image

**To:**
```cs
using FlaxEngine;
using FlaxEngine.GUI;

namespace Game
{
    public class MyControl : Control
    {
        [EditorOrder(0), Tooltip("The Tint color for the texture. Uses red as default.")]
        public Color TintColor { get; set; } = Color.Red;

        [EditorOrder(1), Tooltip("The texture to be drawn.")]
        public Texture Image { get; set; }

        /// <inheritdoc />
        public override void Draw()
        {
            base.Draw();

            Render2D.DrawTexture(Image, new Rectangle(Vector2.Zero, Size), TintColor);
        }
    }
}
```

Overall lots and lots and lots of fixing typos here and there, not trying to nitpick, just want to improve the quality :)

### CODE REVIEW

File: manual/ui/tutorials/create-custom-control.md [hunk 1]
```diff
@@ -1,43 +1,47 @@
-# HOWTO: Create custom control
+# HOWTO: Create a custom control
 
-In this tutorial, you will learn how to create customized User Interface control for your game. Flax Engine uses UI scripted on a C# side so you can add your own controls and container controls as needed.
+In this tutorial, you will learn how to create  a customized User Interface control for your game. Flax Engine uses UI scripted on a C# side so you can add your own controls and container controls as needed.
```
Comment (stefnotch) [inline manual/ui/tutorials/create-custom-control.md:3]: `  a` has one space too many

**Review by stefnotch — APPROVED** [review]
I checked out everything and added a few nitpicky comments.

I do have to say, this whole PR is really great! Thank you so much for taking your time to improve the documentation 🎉

File: manual/ui/control/index.md [hunk 2]
```diff
@@ -2,11 +2,11 @@
 
 ![UI Control](media/title.jpg)
 
-**UI Control** is an actor type that contains a single GUI control. It handles the control linkage to the parent container control or canvas and provides a solid data serialization along with C# API.
+**UI Control** is an actor type that contains a single GUI control. It handles the control linkage to the parent container control or canvas and provides solid data serialization along with the C# API.
 
-## Setting control type
+## Setting up a control type
```
[the remark this answers is not in the document]
Comment (W2Wizard) [reply manual/ui/control/index.md:7]: Done.

File: manual/ui/controls/image.md [hunk 3]
```diff
@@ -2,4 +2,4 @@
 
 ![Image](media/image.png)
 
-The **Image** control displays a non-interactive graphic defined by the assigned brush. It can draw texture, render target, sprite, GUI material, solid color or a linear gradient.
+The **Image** control displays a non-interactive graphic defined by the assigned brush. It can a draw texture, render target, sprite, GUI material, solid color or a linear gradient.
```
[the remark this answers is not in the document]
Comment (W2Wizard) [reply manual/ui/controls/image.md:5]: Oops!

Done.

File: manual/ui/tutorials/create-custom-control.md (hunk 1 above)
Comment (W2Wizard) [reply manual/ui/tutorials/create-custom-control.md:3]: Done.

**Pull request merged**